platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationEntryImpl.java

```diff
@@ -268,24 +272,43 @@ private void handleRestoreWhenAnEntryWasExported(InputStream is, @Nullable Strin
     LOGGER.debug("Importing {}...", debugString);
     try {
       FileUtils.copyInputStreamToFile(is, file);
+      SecurityLogger.audit("Imported file {}", file);
     } catch (IOException e) {
       if (!file.canWrite()) { // make it writable and try again
-        InputStream ris = context.getInputStreamFor(entry);
-
-        try {
-          LOGGER.debug("temporarily overriding write privileges for {}", file);
-          if (!file.setWritable(true)) { // cannot set it writable so bail
-            throw e;
-          }
-          FileUtils.copyInputStreamToFile(ris, file);
-        } finally {
-          IOUtils.closeQuietly(ris); // we do not care if we cannot close it
-          // reset the permissions properly
-          file.setWritable(false);
+        if (!retryHandleRestoreWhenAnEntryWasExported(context.getInputStreamFor(entry))) {
+          throw e;
         }
+      } else {
+        SecurityLogger.audit("Error importing file {}", file);
+        throw e;
       }
     } finally {
       IOUtils.closeQuietly(is); // we do not care if we cannot close it
     }
   }
+
+  private boolean retryHandleRestoreWhenAnEntryWasExported(InputStream is) throws IOException {
+    LOGGER.debug("temporarily overriding write privileges for {}", file);
+    if (file.setWritable(true)) {
+      SecurityLogger.audit("Enabled write privileges for file {}", file);
+    } else { // cannot set it writable so bail
+      SecurityLogger.audit("Error enabling write privileges for file {}", file);
+      return false;
+    }
+    try {
+      FileUtils.copyInputStreamToFile(is, file);
+      SecurityLogger.audit("Imported file {}", file);
+      return true;
+    } catch (IOException ee) {
+      SecurityLogger.audit("Error importing file {}", file);
+      throw ee;
+    } finally {
+      IOUtils.closeQuietly(is); // we do not care if we cannot close it
+      if (file.setWritable(false)) { // reset the permissions properly
```

Comment:
❗️  What if the file was already writeable? According to the java doc (or lack there of):
https://docs.oracle.com/javase/8/docs/api/java/io/File.html#setWritable-boolean-
I think the initial call to `file.setWritable(true)` will succeed, and now we will attempt to revert to an invalid state. I think we need to cache the initial state using `canWrite()` and revert accordingly. 

Unless of course I'm misunderstanding and we **want** to disable write for all files this code touches?

Reply:
private method called internally ... added a comment just for you 😆